core/src/client.rs

```diff
@@ -1,61 +1,31 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! A couple of client types that we use elsewhere.
+
 use crate::{config::Config, metadata::Metadata};
 use derive_where::derive_where;
 
-/// Each client should be able to provide access to the following fields
-/// - runtime version
-/// - genesis hash
-/// - metadata
+/// This provides access to some relevant client state in signed extensions,
+/// and is just a combination of some of the available properties.
 #[derive_where(Clone, Debug)]
 pub struct ClientState<C: Config> {
-    genesis_hash: C::Hash,
-    runtime_version: RuntimeVersion,
-    metadata: Metadata,
-}
-
-impl<C: Config> ClientState<C> {
-    pub fn new(genesis_hash: C::Hash, runtime_version: RuntimeVersion, metadata: Metadata) -> Self {
-        Self {
-            genesis_hash,
-            runtime_version,
-            metadata,
-        }
-    }
-
-    pub fn metadata(&self) -> Metadata {
-        self.metadata.clone()
-    }
-
-    pub fn runtime_version(&self) -> RuntimeVersion {
-        self.runtime_version
-    }
-
-    pub fn genesis_hash(&self) -> C::Hash {
-        self.genesis_hash
-    }
+    /// Genesis hash.
```

Comment:
why pub now?

Reply:
I think is more in-line with `RuntimeVersion`, but either way is good for me

Reply:
It's probably because we use it outside subxt-core but ideally we wouldn't make it pub. It's hard to add stuff without breaking the API...

Reply:
I would probably add doc(hidden) to it to be on the safe-side

Reply:
Yeah, I made runtime version things pub just because it felt better constructing it with named params (and it's basically just a named tuple). I think I followed the same logic here and just made it pub since it's just a tuple of 3 things you can fetch individually otherwise. I could go either way really, but at the time preferred the simplicity I guess of it just being a simple data struct :)

I don't think it should be `doc(hidden)` personally because it's a used as a param in a couple of places so people should be able to name it.